### ISSUE TITLE
Runner doesn't have information about suites

### ISSUE THREAD

**Issue by 1999** [yandex/mocha-parallel-tests#115]
Copied from https://github.com/yandex/mocha-parallel-tests/issues/113#issuecomment-299446681

Okay, I found the reason for this. Runner in mocha and runner in mocha-parallel-tests contain different suites. This is about mocha:

![image](https://cloud.githubusercontent.com/assets/73191/25744590/078d11ca-31de-11e7-882f-d9f134e1b150.png)

And this is about mocha-parallel-tests:

![image](https://cloud.githubusercontent.com/assets/73191/25744608/1c40b6f8-31de-11e7-8a65-36579b7dd367.png)

Probably the reason for this is that suites are all inside runner for mocha and this is kinda different for mocha-parallel-tests. Anyway, this is another bug and it's a blocker for this one. I will create a separate issue for that.

**Comment by 1999** [comment]
It seems like what we need to do is to add suite from lib/watcher.js#scheduleMochaRun to runner singleton instance. The controversial part here is that for mocha all suites are just array members, but for mocha-parallel-tests they run in parallel and it's not clear how we should add them.

Smth like this is a good start:
```diff
diff --git lib/reporter.js lib/reporter.js
index 3a69eab..7796243 100644
--- lib/reporter.js
+++ lib/reporter.js
@@ -213,7 +213,10 @@ class AbstractReporter extends MochaBaseReporter {
 
             if (testsFinished === this._options.testsLength) {
                 debugLog(`This is the end: ${testsFinished}/${this._options.testsLength}`);
-                this._runnerWrapped.end(failsOccured, reporterInstance);
+
+                process.nextTick(() => {
+                    this._runnerWrapped.end(failsOccured, reporterInstance);
+                });
             } else {
                 debugLog(`This is still not the end: ${testsFinished}/${this._options.testsLength}`);
             }
diff --git lib/utils/runner-wrapped.js lib/utils/runner-wrapped.js
index bb64966..eb3b315 100644
--- lib/utils/runner-wrapped.js
+++ lib/utils/runner-wrapped.js
@@ -29,6 +29,10 @@ export default class RunnerWrapped extends MochaRunner {
         this.emit('end', {failsOccured, reporter});
     }
 
+    foobar(suite) {
+        this.suite.suites.push(suite);
+    }
+
     /**
      * Emit events
      *
diff --git lib/watcher.js lib/watcher.js
index a00329b..23898d7 100644
--- lib/watcher.js
+++ lib/watcher.js
@@ -5,6 +5,7 @@ import path from 'path';
 import debug from 'debug';
 import Mocha from 'mocha';
 import RequireCacheWatcher from './utils/require-cache-watcher';
+import {getInstance as getRunnerInstance} from './runner';
 
 const debugLog = debug('mocha-parallel-tests:watcher');
 const cacheWatcher = new RequireCacheWatcher;
@@ -118,6 +119,9 @@ const scheduleMochaRun = ({file, suiteIndex = -1, options, throttledCalls}) => {
         return () => {
             debugLog(`Suite ${suiteHumanIndex} processed (success): ${relativeFile}`);
 
+            const runner = getRunnerInstance();
+            runner.foobar(mocha.suite);
+
             const runnerId = getRunnerId(file, suiteIndex);
             runners.delete(runnerId);
 

```

**Comment by 1999** [comment]
This is fixed in v2
